Website/src/rendering/src/components/Checkout/ShipMethodsList.tsx

```diff
@@ -0,0 +1,49 @@
+import { RequiredDeep, ShipMethodSelection } from 'ordercloud-javascript-sdk';
+import { ChangeEvent } from 'react';
+import { formatCurrency } from '../../helpers/CurrencyHelper';
+import { DShipMethod } from '../../models/ordercloud/DShipMethod';
+
+type ShipMethodsListProps = {
+  shipMethods?: DShipMethod[];
+  shipEstimateId?: string;
+  selectedShipMethodId?: string;
+  onChange?: (shipMethod: RequiredDeep<ShipMethodSelection>) => void;
+};
+
+const ShipMethodsList = (props: ShipMethodsListProps): JSX.Element => {
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const shipMethodId = event.target.value;
+    const shipMethod = props.shipMethods.find((shipMethod) => shipMethod.ID === shipMethodId);
+
+    if (props.onChange && typeof props.onChange === 'function') {
```

Comment:
Is the typeof function check necessary? The props already define onChange as a function.

Reply:
The props define it as an optional parameter. Its useful not having to define in storybook for example. If you prefer I can make it required and update our storybook examples though to do a no-op function.